lib/stack_master/stack_definition.rb

```diff
@@ -44,7 +44,10 @@ def ==(other)
     end
 
     def template_file_path
-      File.expand_path(File.join(template_dir, template))
+      # Use Downloaded file from S3 to do further
+      # operations
+      @template_dir = s3_template_dir if s3.key?("use_remote") && s3["use_remote"] == true
```

Comment:
It would be fairly trivial to abstract this logic into a getter method for `template_dir`:

```
def template_dir
  if s3.fetch("use_remote", false)
    s3_template_dir
  else
    @template_dir
  end
end
```

This will help with any other methods that may want to use (or may already be using) the accessor of this class.